src/MonoTorrent/MonoTorrent.Client/Managers/TorrentManager.cs

```diff
@@ -1001,7 +1001,7 @@ internal async ReusableTask MaybeLoadFastResumeAsync ()
             }
         }
 
-        internal async ReusableTask MaybeWriteFastResumeAsync ()
+        public async ReusableTask MaybeWriteFastResumeAsync ()
```

Comment:
> change MaybeWriteFastResumeAsync to public for Saving Fast Resume file periodic,or before app shutdown due to exception

Perhaps this could be best supported as a setting in `EngineSettings` ? `public enum FastResumeMode { Lossless, Lossy }`.

In lossless mode you are guaranteed that you will never have to redownload data which has passed it's hash check as the fast resume data will only be deleted from disk when a torrent enters 'Downloading' mode and will only be written to disk when the torrent stops, or enters 'seeding' mode. In the event of a crash while downloading you have to hash check the entire file before downloading again.

In 'lossy' mode fast resume data will be stored regularly on disk, but in the event of a crash you will lose the most recent N seconds/minutes of downloaded data as the fast resume file will be slightly stale.

Reply:
how about use Lossy as default value ? usually files are very big, and even worse if there are many files ,

Reply:
https://github.com/alanmcgovern/monotorrent/tree/fast-resume-improvements